### PR TITLE
Garbage Collection - Handle desynced players

### PR DESCRIPTION
**Pull request by Sniperhid**
**When merged this pull request will:**
- Unfortunately vanilla gear desyncs still occur. This PR aims to allow ACRE to continue to function normally if gear desync occurs. 
- Instead of garbage collecting immediately the server sends a global intention to garbage collect a radio. Clients check if the intended radio is in their inventory and if it is the client will send messages on a regular interval to the server to not garbage collect the server. The server will only garbage collect the radio once it does receive a message asking it to after a certain amount of time.
- Removes the gear desync check - it will be made obsolete - RPT logging will still occur if peculiar behaviour does appear. 
- Documents various headers throughout sys_server and also tidies up some of sys_server

### CODE REVIEW

**Comment by Cuel** [comment]
Would it be worth having a config entry to make these visible during play, so the client can grab a new radio?

**Comment by Sniperhid** [comment]
There shouldn't any need as with this PR things should continue to work just fine (that is the point of this), so the player shouldn't notice any changes. It will however log to RPT for debug purposes - it may be indicative of a poor gear allocation system.

File: addons/sys_server/XEH_postInitClient.sqf [hunk 1]
```diff
@@ -4,4 +4,5 @@ NO_DEDICATED;
 
 [QGVAR(clientGCRadio), { _this call FUNC(clientGCRadio) }] call CALLSTACK(CBA_fnc_addEventHandler);
 ["acre_updateIdObjects", { _this call FUNC(updateIdObjects) }] call CALLSTACK(CBA_fnc_addEventHandler);
-[QGVAR(restoreInvalidGCData), { _this call FUNC(cloneRadioData) }] call CALLSTACK(CBA_fnc_addEventHandler);
+
+[QGVAR(intentToGarbageCollect), { _this call FUNC(clientIntentToGarbageCollect); }] call CALLSTACK(CBA_fnc_addEventHandler);
```
Comment (jonpas) [inline addons/sys_server/XEH_postInitClient.sqf]: No semi-colon inside code block.

File: addons/sys_server/XEH_postInitServer.sqf [hunk 2]
```diff
@@ -1,18 +1,17 @@
 #include "script_component.hpp"
 
-["acre_checkServerGearDesync", { _this call FUNC(checkServerGearDesync) }] call CALLSTACK(CBA_fnc_addEventHandler);
-
 LOG("Starting radio id handler events");
 ["acre_getRadioId", { _this call FUNC(onGetRadioId) }] call CALLSTACK(CBA_fnc_addEventHandler);
 ["acre_acknowledgeId", { _this call FUNC(acknowledgeId) }] call CALLSTACK(CBA_fnc_addEventHandler);
+[QGVAR(stopRadioGarbageCollect), { _this call FUNC(stopRadioGarbageCollect); }] call CALLSTACK(CBA_fnc_addEventHandler);
```
Comment (jonpas) [inline addons/sys_server/XEH_postInitServer.sqf]: Semi-colon.

File: addons/sys_server/XEH_postInitServer.sqf [hunk 3]
```diff
@@ -1,18 +1,17 @@
...
+[QGVAR(removeGCQueue), { _this call FUNC(removeGCQueue); }] call CALLSTACK(CBA_fnc_addEventHandler);
```
Comment (jonpas) [inline addons/sys_server/XEH_postInitServer.sqf]: Semi-colon.

File: addons/sys_server/fnc_onGetRadioId.sqf [hunk 4]
```diff
@@ -1,16 +1,18 @@
 /*
  * Author: ACRE2Team
- * SHORT DESCRIPTION
+ * This function is used to start initializing a radio for the intended player. The callback is used to complete to the process.
```
Comment (jonpas) [inline addons/sys_server/fnc_onGetRadioId.sqf]: `to complete to the`?

File: addons/sys_server/fnc_onGetRadioId.sqf [hunk 5]
```diff
@@ -1,16 +1,18 @@
...
  *
  * Arguments:
- * 0: ARGUMENT ONE <TYPE>
- * 1: ARGUMENT TWO <TYPE>
+ * 0: Player's game object <OBJECT>
+ * 1: Radio base classname <TYPE>
+ * 2: CBA event name that is triggered when complete <STRING>
+ * 3: Optional: Replacement ID - Use this when copying data from another radio <STRING>
```
Comment (jonpas) [inline addons/sys_server/fnc_onGetRadioId.sqf]: ` * 3: Replacement ID - Use this when copying data from another radio <STRING> (default: "")`

File: addons/sys_server/fnc_onGetRadioId.sqf [hunk 6]
```diff
@@ -1,16 +1,18 @@
 /*
  * Author: ACRE2Team
- * SHORT DESCRIPTION
+ * This function is used to start initializing a radio for the intended player. The callback is used to complete to the process.
  *
  * Arguments:
- * 0: ARGUMENT ONE <TYPE>
- * 1: ARGUMENT TWO <TYPE>
+ * 0: Player's game object <OBJECT>
+ * 1: Radio base classname <TYPE>
```
Comment (jonpas) [inline addons/sys_server/fnc_onGetRadioId.sqf]: `<STRING>` (Although it's not a lie, string is a type! 😆 )

File: addons/sys_server/fnc_removeGCQueue.sqf [hunk 7]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Author: ACRE2Team
+ * Remove the specified radio from the GC queue. This is typically called when a client no longer has a desynced radio to reset it's GC status.
```
Comment (jonpas) [inline addons/sys_server/fnc_removeGCQueue.sqf]: `its`

File: addons/sys_server/fnc_removeGCQueue.sqf [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * Arguments:
+ * 0: Radio ID <STRING>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [acre_player,"acre_prc152_id_1"] call acre_sys_server_fnc_removeGCQueue
```
Comment (jonpas) [inline addons/sys_server/fnc_removeGCQueue.sqf]: Wrong example, shouldn't have `acre_player` as first argument.

File: addons/sys_server/fnc_updateIdObjects.sqf [hunk 9]
```diff
@@ -1,16 +1,15 @@
 /*
  * Author: ACRE2Team
- * SHORT DESCRIPTION
+ * Client function to update the local copy of objectIdRelationTable
```
Comment (jonpas) [inline addons/sys_server/fnc_updateIdObjects.sqf]: Missing final point.

**Review by jonpas — CHANGES_REQUESTED** [review]

**Review by jonpas — APPROVED** [review]

**Pull request merged**